.github/workflows/build-and-release-ctl.yml

```diff
@@ -28,16 +27,42 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.RELEASE_NOTES_PAT }}
           RELEASE_CHANNEL: "stable"
-          Tag: ${{inputs.release-tag}}
+          TAG: ${{inputs.release-tag}}
         with:
           version: 2
           args: release --clean --skip=validate -f .github/.goreleaser.yml
+
+        # ---------------- Cosign signing step----------------
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@v3.5.0
+
+      - name: Sign mesheryctl artifacts (keyless)
+        run: |
+          for artifact in dist/*; do
+            cosign sign-blob \
+              --yes \
+              --bundle "${artifact}.sigstore" \
+              --output-signature "${artifact}.sig" \
+              --output-certificate "${artifact}.pem" \
+              "$artifact"
+          done
+
+      - name: Verify mesheryctl artifacts
+        run: |
+          for artifact in dist/*; do
+            cosign verify-blob \
+              --certificate-identity-regexp "https://github.com/meshery/meshery/.*" \
+              --certificate-oidc-issuer "https://token.actions.githubusercontent.com" \
+              --signature "${artifact}.sig" \
+              --certificate "${artifact}.pem" \
```

Comment:
we can likely use the --bundle again and same as above remove the sign and cert here too